changeset-feedback/generateFeedback.ts

```diff
@@ -73,57 +74,46 @@ export async function listChangedFiles(ref: string) {
   }
 
   const { stdout } = await execFile('git', ['diff', '--name-only', ref]);
-  return stdout.trim().split(/\r?\n/);
+  return stdout
+    .trim()
+    .split(/\r?\n/)
+    .map((line: string) => resolvePath(process.cwd(), line));
 }
 
-export async function listPackages() {
-  const rootPkg = require(resolvePath(process.cwd(), './package.json'));
-  if (!rootPkg?.workspaces?.packages) {
-    throw new Error('No workspaces found in root package.json');
+const findPackagesInDir = async (dir: string) => {
+  const { packages } = await getPackages(dir).catch(() => ({ packages: [] }));
+  return packages
+    .filter(p => p.relativeDir !== '.')
+    .map(p => ({
+      ...p,
+      relativeDir: relativePath(process.cwd(), resolvePath(dir, p.relativeDir)),
+    }));
+};
+
+export async function listPackages({
+  isCommunityPluginsRepo,
+}: {
+  isCommunityPluginsRepo?: boolean;
+}) {
+  if (!isCommunityPluginsRepo) {
+    return findPackagesInDir(process.cwd());
   }
 
-  const pkgs: Array<{ path: string; name: string }> = [];
+  const workspacesRoot = resolvePath(process.cwd(), 'workspaces');
+  const workspaceDirs = await fs.promises.readdir(workspacesRoot);
 
-  // Naive workspace lookup implementation, we can't shell out to yarn here as the implementation embedded in the repo
-  for (const pkgPath of rootPkg?.workspaces?.packages) {
-    const readDirRecursive = (dir: string, parts: string) => {
-      const [nextPart] = parts;
-
-      // We've reached the end of the path pattern, check if package.json exists
-      if (!nextPart) {
-        try {
-          const pkg = require(resolvePath(dir, 'package.json'));
-          pkgs.push({
-            path: relativePath(process.cwd(), dir),
-            name: pkg.name,
-          });
-        } catch {
-          process.stderr.write(`Failed to read package.json in ${dir}\n`);
-        }
-        return;
-      }
-
-      for (const filePath of fs.readdirSync(dir)) {
-        if (fs.statSync(resolvePath(dir, filePath)).isDirectory()) {
-          if (filePath === nextPart || nextPart === '*') {
-            readDirRecursive(resolvePath(dir, filePath), parts.slice(1));
-          }
-        }
-      }
-    };
-
-    // Split the workspace paths by / and check each directory level recursively
-    readDirRecursive(process.cwd(), pkgPath.split('/'));
-  }
-
-  return pkgs;
+  return await Promise.all(
+    workspaceDirs.map(workspace =>
+      findPackagesInDir(resolvePath(workspacesRoot, workspace)),
+    ),
+  ).then(packages => packages.flat());
 }
 
 export async function loadChangesets(filePaths: string[]) {
   const changesets = [];
   for (const filePath of filePaths) {
     if (
-      !filePath.startsWith('.changeset/') ||
+      !filePath.includes('.changeset/') ||
       !filePath.endsWith('.md') ||
       filePath.endsWith('README.md')
```

Comment:
Could make sense to bring in `glob` or something here too

Reply:
yeh, it's a little tricky though as we already have the files that have changed in an array. Think I'll leave this for a later refactor :)